x-pack/plugins/security_solution/public/common/components/toolbar/bulk_actions/alert_bulk_tags.tsx

```diff
@@ -0,0 +1,170 @@
+import { getUpdateAlertsQuery } from '../../../../detections/components/alerts_table/actions';
+import { DEFAULT_ALERT_TAGS_KEY } from '../../../../../common/constants';
+import { useUiSetting$ } from '../../../lib/kibana';
+import { useSetAlertTags } from './use_set_alert_tags';
+import { useAppToasts } from '../../../hooks/use_app_toasts';
+import * as i18n from './translations';
+import { createInitialTagsState } from './helpers';
+
+interface BulkAlertTagsPanelComponentProps {
+  alertItems: TimelineItem[];
+  refetchQuery?: () => void;
+  setIsLoading: (isLoading: boolean) => void;
+  refresh?: () => void;
+  clearSelection?: () => void;
+  closePopoverMenu: () => void;
+}
+const BulkAlertTagsPanelComponent: React.FC<BulkAlertTagsPanelComponentProps> = ({
+  alertItems,
+  refresh,
+  refetchQuery,
+  setIsLoading,
+  clearSelection,
+  closePopoverMenu,
+}) => {
+  const [defaultAlertTagOptions] = useUiSetting$<string[]>(DEFAULT_ALERT_TAGS_KEY);
+  const { addSuccess, addError, addWarning } = useAppToasts();
+
+  const { setAlertTags } = useSetAlertTags();
+  const existingTags = useMemo(
+    () =>
+      alertItems.map(
+        (item) => item.data.find((data) => data.field === ALERT_WORKFLOW_TAGS)?.value ?? []
+      ),
+    [alertItems]
+  );
+  const initalTagsState = useMemo(
+    () => createInitialTagsState(existingTags, defaultAlertTagOptions),
+    [existingTags, defaultAlertTagOptions]
+  );
+
+  const tagsToAdd: Set<string> = useMemo(() => new Set(), []);
+  const tagsToRemove: Set<string> = useMemo(() => new Set(), []);
+
+  const onUpdateSuccess = useCallback(
+    (updated: number, conflicts: number) => {
+      if (conflicts > 0) {
+        addWarning({
+          title: i18n.UPDATE_ALERT_TAGS_FAILED(conflicts),
+          text: i18n.UPDATE_ALERT_TAGS_FAILED_DETAILED(updated, conflicts),
+        });
+      } else {
+        addSuccess(i18n.UPDATE_ALERT_TAGS_SUCCESS_TOAST(updated));
+      }
+    },
+    [addSuccess, addWarning]
+  );
+
+  const onUpdateFailure = useCallback(
+    (error: Error) => {
+      addError(error.message, { title: i18n.UPDATE_ALERT_TAGS_FAILURE });
+    },
+    [addError]
+  );
+
+  const [selectableAlertTags, setSelectableAlertTags] =
+    useState<EuiSelectableOption[]>(initalTagsState);
+
+  const onTagsUpdate = useCallback(async () => {
+    closePopoverMenu();
+    const ids = alertItems.map((item) => item._id);
+    const query: Record<string, unknown> = getUpdateAlertsQuery(ids).query;
+    const tagsToAddArray = Array.from(tagsToAdd);
+    const tagsToRemoveArray = Array.from(tagsToRemove);
+    try {
+      setIsLoading(true);
+
+      const response = await setAlertTags({
+        tags: { tags_to_add: tagsToAddArray, tags_to_remove: tagsToRemoveArray },
+        query,
+      });
+
+      setIsLoading(false);
+      if (refetchQuery) refetchQuery();
+      if (refresh) refresh();
+      if (clearSelection) clearSelection();
```

Comment:
Can we group all these props under a single `onUpdateSuccess` prop? And let the caller do whatever it needs when the update succeeded, outside this component.